request.cc

```diff
@@ -0,0 +1,120 @@
+#include "request.h"
+
+#include <boost/tokenizer.hpp>
+#include <boost/algorithm/string.hpp>
+
+using Headers = std::vector<std::pair<std::string, std::string>>;
+
+// Helper function to tokenize a string
+boost::tokenizer<boost::char_separator<char>> 
+token_generator(const std::string s, const char* sep) {
+  // Tokenize the rest_lines 
+  boost::char_separator<char> separator(sep);
+  boost::tokenizer<boost::char_separator<char>> tokens(s, separator);
+  return tokens;
+}
+
+std::unique_ptr<Request> Request::Parse(const std::string& raw_request) {
+  std::unique_ptr<Request> req = std::unique_ptr<Request>(new Request);
+  req->raw_request_ = raw_request;
+
+  // No support for POST requests yet, so set body_ to empty string
+  req->body_ = "";
+
+  int pos = raw_request.find_first_of('\r\n');
+
+  std::string first_line = raw_request.substr(0, pos);
+
+  std::string rest_lines = raw_request.substr(pos+1);
+  std::cout << "first_line: " + first_line << std::endl;
+  std::cout << "rest_lines: " + rest_lines << std::endl;
+
+  // Replace '\r\n' with space for tokenizing later
```

Comment:
This comment should be updated to say replace `\r\n` with `~~~` or update lines 34 and 35 to reflect the comment.